subgraph/src/utils.ts

```diff
@@ -15,3 +24,39 @@ export function buildAuthorizedSignerId(
   );
   return Bytes.fromByteArray(hash);
 }
+
+/**
+ * Generate a keccak256 hex string of the user:eventType:blockTimestamp
+ * @param user the user the event belongs to/performed the action
+ * @param eventType the UseSubscriptionEventType enum value
+ * @param blockTimestamp the block timestamp of the event
+ * @returns Bytes representation of a hex string concatenation of the `user:eventType:blockTimestamp` to create a unique id
+ */
+export function buildUserSubscriptionEventId(
+  user: Bytes,
+  eventType: string,
+  blockTimestamp: BigInt
```

Comment:
I went back and forth on using the block timestamp vs the transaction hash. Thoughts?

Reply:
I also prefer the timestamp in general. This seems fine since the contract limits a user to 1 active subscription, and any valid combination of subscribe/unsubscribe events that can occur within a single block (and therefore at the same timestamp) will be treated as one of these new user subscription events.